kokiri/src/graphics/opengl/renderer2d.cpp

```diff
@@ -6,16 +6,56 @@
 namespace Kokiri {
     namespace Graphics {
         namespace OpenGL {
-            Renderer2D::Renderer2D(Window &&window) {
-                m_context = new Context(std::move(window), Context::ContextType::CORE);
+            Renderer2D::Renderer2D(Core::Shared<Core::Window> window) {
+                Renderer2D(window, Version::OPENGL_4_6);
             }
 
-            Renderer2D::Renderer2D(Window &&window, Context::ContextVersion version) {
-                m_context = new Context(std::move(window), Context::ContextType::CORE, version);
+            Renderer2D::Renderer2D(Core::Shared<Core::Window> window, Version version) {
+
+                auto type = Type::CORE;
+
+                int major = 2;
```

Comment:
@rafaelcn create a function that returns a tuple with openGL major and minor versions, it's too much noisy mixup this code in the constructor here. Something like that:

```cpp
std::pair<int, int> getOpenGLVersion()
{
    int majorVersion, minorVersion;
    glGetIntegerv(GL_MAJOR_VERSION, &majorVersion);
    glGetIntegerv(GL_MINOR_VERSION, &minorVersion);
    return std::make_pair(majorVersion, minorVersion);
}
```

https://stackoverflow.com/questions/24708231/how-to-find-my-opengl-version-in-linuxubuntu-14-04

Reply:
I think I have a better idea... let me cook it and update this code, it surely is much better than these amount of if/else conditions.